### PR TITLE
Fix various nightly clippy warnings

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: glib-macros/tests/value_delegate_derive.rs [hunk 1]
```diff
@@ -31,10 +31,10 @@ fn into_value() {
         }
     }
 
-    test_func(&Test(123));
+    test_func(Test(123));
     test_func(Test(123));
```
Comment (sdroege) [inline glib-macros/tests/value_delegate_derive.rs:35]: This is defeating the point of the test :D It should check that both values and references work

File: glib/src/boxed.rs [hunk 2]
```diff
@@ -330,7 +330,7 @@ macro_rules! glib_boxed_wrapper {
         impl $(<$($generic $(: $bound $(+ $bound2)*)?),+>)? $crate::prelude::StaticType for $name $(<$($generic),+>)? {
             #[inline]
             fn static_type() -> $crate::types::Type {
-                #[allow(unused_unsafe)]
+                #[allow(macro_metavars_in_unsafe)]
```
[the remark this answers is not in the document]
Comment (sdroege) [reply glib/src/boxed.rs:333]: What does it do? Can we just ignore it for now and hope it goes not into stable?

**Review by sdroege — DISMISSED** [review]
Looks good to me otherwise. Thanks!

File: glib/src/boxed.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply glib/src/boxed.rs]: > What does it do? 

See https://rust-lang.github.io/rust-clippy/master/index.html#/macro_metavars_in_unsafe

I reverted those changes, we can revise once either patch makes it into stable or not.

**Pull request merged**